File: inspect_excel.js
import XLSX from 'xlsx';
import fs from 'fs';

const file = 'd:/Template/russian-lang/violations_2026-03-06.xlsx';
if (!fs.existsSync(file)) {
    console.log("File not found");
    process.exit(1);
}

const workbook = XLSX.readFile(file);
const sheet = workbook.Sheets[workbook.SheetNames[0]];
const data = XLSX.utils.sheet_to_json(sheet);

const types = data.reduce((acc, r) => {
    const type = r['Тип'];
    acc[type] = (acc[type] || 0) + 1;
    return acc;
}, {});

console.log(JSON.stringify({
    total_rows: data.length,
    types_summary: types,
    sample: data.slice(0, 5)
}, null, 2));

File: inspect_excel.py
import pandas as pd
import json
import os

file_path = 'd:/Template/russian-lang/violations_2026-03-06.xlsx'
if not os.path.exists(file_path):
    print(f"Error: File {file_path} not found")
else:
    try:
        df = pd.read_excel(file_path)
        info = {
            "columns": df.columns.tolist(),
            "row_count": len(df),
            "sample": df.head(15).to_dict(orient='records'),
            "types_summary": df['Тип'].value_counts().to_dict() if 'Тип' in df.columns else "Type column missing"
        }
        print(json.dumps(info, ensure_ascii=False, indent=2))
    except Exception as e:
        print(f"Error reading file: {e}")
